include/ensmallen_bits/callbacks/timer_stop.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file timer_stop.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of the timer stop callback function.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_CALLBACKS_TIMER_STOP_HPP
+#define ENSMALLEN_CALLBACKS_TIMER_STOP_HPP
+
+namespace ens {
+
+/**
+ * Timer stop function, based on the EndEpoch callback function.
+ */
+class TimerStop
+{
+ public:
+  /**
+   * Set up the print loss callback class with the width and output stream.
+   *
+   * @param time The timeout in seconds.
+   */
+  TimerStop(const double time) : time(time), reset(false)
```

Comment:
I suggest to disambiguate the name of the input variable from the name of the class member variable. Example:

    TimerStop(const double in_time) : time(in_time), reset(false)

Otherwise certain lower-quality compilers may get confused about which variable we're referring to, and we can end up with `time` not being initialised at all.

When in doubt, assume the compiler will do the wrong thing :)